src/ucs/sys/sock.h

```diff
@@ -130,11 +133,13 @@ int ucs_socket_max_conn();
  * @param [in/out]  length_p    The length, in bytes, of the data in buffer
  *                              pointed to by the `data` parameter. The amount of
  *                              data transmitted is written to this argument.
+ * 
  *
  * @return UCS_OK on success, UCS_ERR_CANCELED if connection closed,
  *         UCS_ERR_IO_ERROR on failure.
  */
-ucs_status_t ucs_socket_send_nb(int fd, const void *data, size_t *length_p);
+ucs_status_t ucs_socket_send_nb(int fd, const void *data, size_t *length_p,
+                                ucs_socket_io_err_handler_t *io_err_handler);
```

Comment:
i think we can pass err_cb and err_arg as separate arguments, without uniting them to one sstructure

Reply:
done